### PR TITLE
Implement `UnixDatagram::connect()`

### PR DESCRIPTION
**Pull request by tailhook**
I think this belongs here.

My primary use case for now is `syslog`. I think it's bad to use `send_to` to send each time to same socket. I think it's a little bit slower because address is passed to kernel every time.

### CODE REVIEW

**Comment by tailhook** [comment]
@sfackler any issues with this?

This is a blocker for the sane implementation of syslog logger (https://github.com/Geal/rust-syslog/issues/1)

File: src/lib.rs [hunk 1]
```diff
@@ -688,6 +688,22 @@ impl UnixDatagram {
         })
     }
 
+    /// Creates a Unix Datagram socket which is connected to specified addresss
+    /// the socket is unnamed and similar to one created by `new()` except
+    /// it will send message to the specified addresss *by default*
+    pub fn connect<P: AsRef<Path>>(path: P) -> io::Result<UnixDatagram> {
```
Comment (sfackler) [inline src/lib.rs]: It seems like this should be an instance method that allows an existing socket to connect.

**Comment by sfackler** [comment]
I think this makes sense, but `recv` should be added as well to complement `send` since a `connect`ed socket can only receive from the socket it's connected to.

**Comment by tailhook** [comment]
Okay, added with separate commits. However:

> I think this makes sense, but recv should be added as well to complement send since a connected socket can only receive from the socket it's connected to.

I don't see how this is related. I.e. I don't know the way to send something **to** the connected socket. Because there is no address to `send_to` or to `connect` to :)

But `recv()` is useful as `recvfrom()` that doesn't care about address (i.e. for bound socket expecting anonymous connections).

**Comment by sfackler** [comment]
Thanks, merged!

There is a way to send to it now that `connect` is an instance method and so can be created from a bound socket.

**Comment by sfackler** [comment]
I'll publish when I get home tonight.

**Pull request merged**